File: 基礎題庫/Contents/c561/c561.c
#include<stdio.h>

int main(void)
{
	int n, num;
	
	while(scanf("%d", &n) != EOF)
	{
		int max = 0;
		
		for(int i = 0; i < n; i++)
		{
			int rev = 0;
			
			scanf("%d", &num);
			while(num > 0)
			{
				rev = rev * 10 + num % 10;
				num /= 10;
			}
			
			if(rev > max)
				max = rev;
		}
		
		printf("%d\n", max);
	}
	
	return 0;
}

File: 基礎題庫/Contents/c561/c561.py
import sys

def rev(x):
    x = str(x)
    x = x[::-1]
    return int(x)

for n in sys.stdin:
    
    temp = [int(e) for e in input().split()]
    data = []
    for num in temp:
        num = rev(num)
        data.append(num)
    
    data = sorted(data)
    print(data[-1])
